src/lib/containers/entity/SchemaDrivenAnnotationEditor/AnnotationEditorUtils.ts

```diff
@@ -95,3 +103,65 @@ export function transformErrors(errors: AjvError[]): AjvError[] {
   // Return the transformed errors.
   return errors
 }
+
+/**
+ * Custom annotations in Synapse are always arrays. This function converts initial data to be an array type.
+ * If the initial data is an array, return the data itself.
+ * If the initial data is a string, returns an array of substrings separated by commas.
+ * Otherwise, wrap the data in an array.
+ */
+export function convertToArray<T>(value: T): Array<any> {
+  if (Array.isArray(value)) {
+    return value
+  } else if (typeof value === 'string') {
+    return value.split(',').map(s => s.trim()) // split a string of comma-separated values, then trim whitespace
+  } else {
+    return [value]
+  }
+}
+
+/**
+ * `componentDidUpdate` function for RJSF ObjectField.
+ *
+ * For an object, this will
+ * - convert additionalProperties formData to arrays
+ * - convert schema-defined formData from an array to a non-array if the schema type is not an array
+ * @param props
+ */
+export function objectFieldComponentDidUpdate(props: FieldProps) {
```

Comment:
This logic used to be split up in `FieldTemplate` and `AdditionalPropertiesSchemaField`. Each of those components wrapped the call to `onChange` in `useEffect` with a `setTimeout`.

That wasn't working very well. In short, multiple fields would call `onChange` at once, and only one would succeed.

This eliminates the race condition by changing all of the fields in the object in one pass.